src/selectors/getSortedRank.ts

```diff
@@ -30,6 +41,16 @@ const getSortedRank = (state: State, id: ThoughtId, value: string) => {
   const isDescending = sortPreference.direction === 'Desc'
   const thoughts = children.filter(thought => !state.cursor || thought.id !== state.cursor[state.cursor.length - 1])
 
+  let editedThought: Thought | undefined = undefined
+  // Find the thought that matches the value being sorted
+  // If multiple thoughts have the same value, use the most recently updated one
+  // This handles edge cases where duplicate values might exist
+  for (const thought of thoughts) {
+    if (thought.value === value) {
+      editedThought = (editedThought ? editedThought.lastUpdated : 0) > thought.lastUpdated ? editedThought : thought
+    }
+  }
```

Comment:
This style of coding doesn't fit the project very well. As you know, we follow a functional programming paradigm. Please avoid mutations and for loops when there are purer approaches.